File: upwork-intelligence/append-searches.mjs
#!/usr/bin/env node
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const DIR = path.dirname(fileURLToPath(import.meta.url));
const main = path.join(DIR, 'search-raw.json');
const addPath = process.argv[2];
if (!addPath) process.exit(1);
const mainData = JSON.parse(fs.readFileSync(main, 'utf8'));
const adds = JSON.parse(fs.readFileSync(addPath, 'utf8'));
const seen = new Set((mainData.searches || []).map((s) => s.keyword));
for (const s of adds.searches || adds) {
  if (!s.keyword || seen.has(s.keyword)) continue;
  mainData.searches.push(s);
  seen.add(s.keyword);
}
if (adds.errors) {
  mainData.errors = [...new Set([...(mainData.errors || []), ...adds.errors])];
}
fs.writeFileSync(main, JSON.stringify(mainData));
console.log(JSON.stringify({ total: mainData.searches.length }));

File: upwork-intelligence/_append_searches.py
#!/usr/bin/env python3
import json
import sys
from pathlib import Path

BASE = Path(__file__).resolve().parent
raw = BASE / "_raw_run.json"
data = json.loads(raw.read_text())
items = json.loads(sys.argv[1] if len(sys.argv) > 1 else sys.stdin.read())
if isinstance(items, dict) and "keyword" in items:
    items = [items]
data["searches"].extend(items)
data["meta"]["keywordsCompleted"] = len(data["searches"])
raw.write_text(json.dumps(data))
print(len(data["searches"]))
